api/apps/api/src/modules/projects/planning-areas/admin-planning-areas.repository.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { assertDefined } from '@marxan/utils';
+import { AdminArea } from '@marxan/admin-regions';
+import { AdminAreasService } from '@marxan-api/modules/admin-areas/admin-areas.service';
+import { notFound, NotFound } from './abstract-planning-areas.service';
+
+@Injectable()
+export class AdminPlanningAreasRepository {
+  constructor(private readonly adminAreasService: AdminAreasService) {}
+
+  async findAdminAreaIdByLevelId(adminAreaId: string) {
+    const foundId:
+      | Pick<AdminArea, 'id'>
+      | NotFound = await this.adminAreasService
+      .getByLevel1OrLevel2Id(adminAreaId, {
+        fields: ['id'],
+      })
```

Comment:
shouldn't it catch there with some other type than `not found` ?

Reply:
I am not sure, I do not expect any other error here so I'm not even sure how should we react to it

Reply:
we explicitly only throw NotFoundException, then of course stuff can happen in TypeORM or whatnot, but I'd say this is ok as is.